tools/check_links.py

```diff
@@ -21,47 +21,72 @@
 
 import collections
 import pathlib
+import requests
 import urllib.parse
 
 import click
 import marko
 
+BAD_STATUS_CODES = [401, 401, 403, 404, 500]
 BASEDIR = pathlib.Path(__file__).resolve().parents[1]
 DOC = collections.namedtuple('DOC', 'path relpath links')
 LINK = collections.namedtuple('LINK', 'dest valid')
 
 
-def check_docs(dir_name):
-  'Traverse dir_name and check links in Markdown files.'
+def check_link(link, readme_path, external):
+  'Checks if a link element has a valid destination.'
+  link_valid = None
+  url = urllib.parse.urlparse(link.dest)
+  # If the link is public, say the link is anyway valid
+  # if --external is not set; check the link otherwise
+  print(link.dest)
+  if url.scheme:
+    link_valid = True
+    if external:
+      user_agent = {'User-Agent': 'Chrome/51.0.2704.64'}
+      try:
+        status_code = requests.get(link.dest, headers=user_agent).status_code
+        if status_code in BAD_STATUS_CODES:
```

Comment:
1. Why change the user agent?
2. Use the [response's ok attribute](https://docs.python-requests.org/en/master/api/#requests.Response.ok)
```
response = requests.get(dest)
link_valid = response.ok
```

Reply:
Done